client/components/review-form.js

```diff
@@ -0,0 +1,80 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+// import { withRouter, NavLink } from 'react-router-dom';
+import { postReview } from '../store/review'
+
+const initialState = {
+  rating: 5,
+  body: ''
+}
+
+export class ReviewForm extends Component {
+  constructor(props){
+    super(props);
+    this.state = initialState;
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  render () {
+
+    return (
+      <div>
+        <h2> Review this sauce </h2>
+        <form onSubmit = {this.handleSubmit}>
+          <label>Rating: </label>
+          <input
+            name = "rating"
+            onChange = {this.handleChange}
+            value = {this.state.rating }
+            placeholder = "Rating"
+            type = "number"
+            min = "1"
+            max = "5"
+            required />
+          <label>Body: </label>
+          <input
+            name = "body"
+            onChange = {this.handleChange}
+            value = {this.state.body}
+            placeholder = "Text here"
+            type = "text"
+            minLength = "10"
+            required />
+          <button type = "submit"> Add Review </button>
+        </form>
+        <h2>Reviews:</h2>
+      </div>
+    )
+  }
+  handleSubmit(event){
+    event.preventDefault();
+    const productId = Number(this.props.user.match.params.productId);
```

Comment:
Just curious - why are you using this to find productId when you could be mapping state to props and filtering?